src/Contract/UnbalancedTx.purs

```diff
@@ -223,6 +267,46 @@ explainMkUnbalancedTxError = case _ of
       )
     <> "\n"
 
+  prettyTokenName :: TokenName -> String
+  prettyTokenName = fromTokenName byteArrayToHex identity
+
+  prettyPlutusScript :: PlutusScript -> String
+  prettyPlutusScript (PlutusScript (code /\ lang)) =
+    show lang <> ": " <> byteArrayToHex code
+
+  prettyTxIn :: TransactionInput -> String
+  prettyTxIn (TransactionInput ti) =
+    "Id: "
+      <> byteArrayToHex (unwrap ti.transactionId)
+      <> "\nIndex: "
+      <>
+        show ti.index
+
+  prettyNativeScript :: Int -> NativeScript -> String
+  prettyNativeScript indent script =
+    let
+      newIndent = indent + 1
+    in
+      case script of
+        ScriptPubkey kh -> rawBytesToHex $ ed25519KeyHashToBytes kh
+        ScriptAll scripts -> "All of:\n" <>
+          joinWithIndentNewline newIndent scripts
+        ScriptAny scripts -> "At least one of:\n" <>
+          joinWithIndentNewline newIndent scripts
+        ScriptNOfK n scripts -> "At least " <> show n <> " of:\n" <>
+          joinWithIndentNewline newIndent scripts
+        TimelockStart slot -> "Timelock start for slot " <> show (unwrap slot)
+        TimelockExpiry slot -> "Timelock expiry for slot " <> show (unwrap slot)
+
+  joinWithIndentNewline :: Int -> Array NativeScript -> String
+  joinWithIndentNewline indent = map (prettyNativeScript indent) >>>
+    String.joinWith ("\n" <> spaces (2 * indent))
+
+  spaces :: Int -> String
+  spaces n
+    | n <= 0 = ""
+    | otherwise = " " <> spaces (n - 1)
+
```

Comment:
These should be domain functions. Please move them to appropriate modules.

Do not re-invent pretty-printing machinery: indentation is automatically supported by `TagSet` stuff. Look how other `pprint*` functions are implemented: the structure is explicitly constructed and generic pretty-printer handles indents.

Rename `pretty` -> `pprint` for consistency with other such functions.